podman/domain/containers_create.py

```diff
@@ -379,7 +378,18 @@ def create(
             headers={"content-type": "application/json"},
             data=payload,
         )
-        response.raise_for_status(not_found=ImageNotFound)
+        if response.status_code == requests.codes.not_found:
+            self.podman_client.images.pull(
+                image,
+                auth_config=kwargs.get("auth_config"),
+                platform=kwargs.get("platform"),
+                policy=kwargs.get("policy", "missing"),
+            )
+            response = self.client.post(
+                "/containers/create",
+                headers={"content-type": "application/json"},
+                data=payload,
+            )
```

Comment:
I was curious to see what podman cli does when you call `create`, so I tried. when you do `podman-remote create` you get a pull call no matter if the image is cached or not

```
POST /v5.7.0/libpod/images/pull
POST /v5.7.0/libpod/containers/create
```

therefore I would simply remove the if and try to pull every time you create